### ISSUE TITLE
Achievement Unlocked 2: Missing right click achievements.

### ISSUE THREAD

**Issue by waspennator** [ruffle-rs/ruffle#10287]
### Describe the bug

https://www.newgrounds.com/portal/view/553791?emulate=flash
Game has 2 achievements specifically linked towards right clicking and clicking on 2 buttons that pop up to unlock them, but on ruffle web they are not showing up at all. It also makes the game impossible on desktop since you can't even right click to open the menu on there.

The achievements in question are these 2
129: Not left click (Right Click, click here achievement)
193: Won't follow direction (Right click, don't click here)

I also checked out Achievement Unlocked 3 and it had no issue with the right click menu showing the extra free achievement button on there least on web, so it seems like it's happening mostly with this game.

![AC2 Ruffle 1](https://user-images.githubusercontent.com/78768499/226452571-f8e5a1fd-e691-4d65-97c3-e03ae565593b.png)

### Expected behavior

![AC2 Flash 1](https://user-images.githubusercontent.com/78768499/226452599-f9a90d50-1f11-4d54-8bbf-b81ac862ea6d.png)

### Affected platform

Browser's extension

### Operating system

Windows 11

### Browser

Edge 110

**Comment by n0samu** [comment]
> I also checked out Achievement Unlocked 3 and it had no issue with the right click menu showing the extra free achievement button on there least on web

Yes, but clicking that button does nothing because AS3 context menu callbacks are not implemented. That's an important detail.

**Comment by waspennator** [comment]
Update, I retested on recent desktop nightly, it's not letting me pull up the context menu period on desktop, though I can on web it seems.

**Comment by waspennator** [comment]
Still happening with latest nightly.

**Comment by n0samu** [comment]
The 2 achievements mentioned in this issue are 129 ("Not Left Click") and 193 ("Won't Follow Directions"), both of which need custom context menu items and context menu callbacks to be implemented in AVM2.
![image](https://github.com/ruffle-rs/ruffle/assets/71368227/9068b146-95e8-4c88-9166-e971d98356a7)

**Comment by applesauceisgod** [comment]
this is sad  wanted to beat this game😢

**Comment by burgerhex** [comment]
Still experiencing this on 0.1.0 nightly build 2025-06-20 on [Armor Games](https://armorgames.com/play/6561/achievement-unlocked-2). But, this issue seems to have been marked as solved in 2024 as part of #17022. Any updates?

**Comment by n0samu** [comment]
@burgerhex AS3 context menu callbacks are a separate feature from right mouse button events. Context menu callbacks still are not implemented.
